File: SolveByC++/Problem1.cpp
/* To run, copy the whole thing into Main.cpp */

#include <iostream>
#include <vector>
using namespace std;

namespace Leetcode
{
	class Problem1
	{
	public:
		static void run()
		{
			// Prepare test input
			vector<int> nums = vector<int>{ 2, 7, 11, 15 };
			int target = 9;

			// Get output
			vector<int> result = twoSum(nums, target);
		}

		static vector<int> twoSum(vector<int>& nums, int target)
		{
			// Solution goes here ...

			return vector<int>{1, 0};
		}
	};
}

File: SolveByCSharp/Leetcode/Problems100/Problem1.cs
using System.Collections.Generic;

namespace SolveByCSharp.Leetcode.Problems100
{
    public static class Problem1
    {
        public static void Run()
        {
            var result = TwoSum(new int[] { 2, 7, 11, 15 }, 9);
        }

        public static int[] TwoSum(int[] nums, int target)
        {
            System.Collections.Generic.Dictionary<int, int> dict = new Dictionary<int, int>();

            for (int i = 0; i < nums.Length; ++i)
            {
                if (dict.ContainsKey(target - nums[i]))
                {
                    return new int[] { i, dict[target - nums[i]] };
                }

                if (!dict.ContainsKey(nums[i]))
                        dict.Add(nums[i], i);
            }
            return new int[] { 0, 0 };
        }
    }
}
